src/core/providers/ogr/qgsogrprovider.cpp

```diff
@@ -3652,6 +3671,9 @@ void QgsOgrProvider::open( OpenMode mode )
 
 void QgsOgrProvider::close()
 {
+  if ( mWriteAccess && mForceRecomputeExtent )
+    extent();
```

Comment:
Can you explain this addition?

Reply:
without it, a failure in testGeopackageExtentUpdate() would appear with the new behaviour when doing
```
        # Test moving a geometry that touches the bbox
        self.assertTrue(vl.startEditing())
        self.assertTrue(vl.changeGeometry(1, QgsGeometry.fromWkt('Point (0.5 0)')))
        self.assertTrue(vl.commitChanges())
        reference = QgsGeometry.fromRect(QgsRectangle(0.5, 0.0, 1.0, 1.0))
        provider_extent = QgsGeometry.fromRect(vl.extent())
        self.assertTrue(QgsGeometry.compare(provider_extent.asPolygon()[0], reference.asPolygon()[0], 0.00001),
                        provider_extent.asPolygon()[0])
```
The reason is that vl.changeGeometry() sets the mForceRecomputeExtent flag, but that one was only taken into extent() to actually trigger the "RECOMPUTE EXTENT ON" command.  With GDAL < 3.4.2, commitChanges() wouldn't close the file, which was in update mode, and thus extent() could complete succesfully. With GDAL >= 3.4.2, commitChanges() causes a close() and reopen(ReadOnly), and thus in close() we need to make sure to force actual extent recomputation.